Tensile/LibraryIO.py

```diff
@@ -410,3 +413,49 @@ def createLibraryLogicList(schedulePrefix, architectureName, deviceNames, logicT
 
     data.append(logicTuple[7])
     return data
+
+
+def readAsmCapsCache(config: dict, args: Namespace) -> dict:
+    """
+    If requested and exists, reads an asm cache from disk.
+
+    Arguments:
+      config: dict containing the GlobalParameters key: value pairs from the Tensile config file
+      args:   Command line arguments.
+
+    Returns:
+      newcache:  If file exists, a dictionary containing capabilities cache; otherwise
+                 an empty dict.
+      cacheFile: The user specified yaml file that contains or will contain the asm cache.
+    """
+    if "GlobalParameters" in config and "AsmCacheFile" in config["GlobalParameters"]:
+      globalParameters["CacheAsmCaps"] = True
+      cacheFile = config["GlobalParameters"]["AsmCacheFile"]
+      globalParameters["AsmCacheFile"] = cacheFile
+    elif args.AsmCacheFile:
+      globalParameters["CacheAsmCaps"] = True
+      cacheFile = args.AsmCacheFile
+      globalParameters["AsmCacheFile"] = cacheFile
+    else:
+      globalParameters["CacheAsmCaps"] = False
+      cacheFile = ""
+
+    useCache = globalParameters["CacheAsmCaps"]
+    cacheExists = os.path.exists(cacheFile)
+
+    cache  = readYAML(cacheFile) if useCache and cacheExists else {}
+    toList = lambda s: tuple([int(i.strip()) for i in s.split(",")])
+    newcache = {toList(k): v for k,v in cache.items()}
+
+    return (newcache, cacheFile)
+
+def writeAsmCapsCache(cacheFile: str, data: dict):
+    """
+    Writes an asm cache to disk.
+
+    Arguments:
+      cacheFile: User spedified file to write the cache.
+      data:      dict containg the computed cache.
+    """
+    newcache = {", ".join(map(str, k)): v for k, v in data.items()}
+    writeYAML(cacheFile, newcache)
```

Comment:
Would JSON be appropriate here? Last I checked, the json.load was significantly faster than yaml.load. I suppose the loading time is probably not significant if this is a small file.

Reply:
I 100% agree but we would have to take on another dependency and loading time is negligible for these files. I have looked at using json for the logic files and it does help but when we first started measuring a few months ago the cost was negligible compared to other operations. That may no longer be the case given the current state of the library. We are now using CSafeLoader which also improved yaml load times.